### PR TITLE
LPS-195821 Create validation to disallow add, delete and edit System Object Validations

### PR DESCRIPTION
**Pull request by pedro-oliveira446**
[LPS-195821](https://liferay.atlassian.net/browse/LPS-195821)

### CODE REVIEW

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Comment by pedro-oliveira446** [comment]
ci:stop

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java [hunk 1]
```diff
@@ -562,6 +571,16 @@ else if (Objects.equals(
 		}
 	}
 
+	private void _validateInvokerBundle(String messageType, boolean system)
```
Comment (gabrielwas) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java:574]: @pedro-oliveira446 Can you put it inside the ObjectDefinitionUtil and consume from there?

Comment (gabrielwas) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java:574]: Remember to adapt the code according to Brian suggestions

**Review by gabrielwas — CHANGES_REQUESTED** [review]
Just that validate change, @pedro-oliveira446 ! Thank you!

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java [hunk 2]
```diff
@@ -569,12 +574,18 @@ public ObjectDefinition putObjectDefinition(
 				serviceBuilderObjectField ->
 					serviceBuilderObjectField.isMetadata() ||
 					!serviceBuilderObjectField.isSystem());
+
```
Comment (gabrielwas) [inline modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java:577]: @pedro-oliveira446 Is it possible to remove this blank line?

**Review by gabrielwas — CHANGES_REQUESTED** [review]

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Review by gabrielwas — APPROVED** [review]
LGTM ;) Good job, @pedro-oliveira446

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectValidation/EditObjectValidation.tsx [hunk 3]
```diff
@@ -93,15 +93,17 @@ export default function EditObjectValidation({
...
+				const {fieldName, message: detailMessage} = JSON.parse(
+					detail as string
+				) as {
+					fieldName: keyof ObjectValidationErrors;
+					message: string;
+				};
 
+				setErrorMessage({[fieldName]: detailMessage});
+			}
 			openToast({message, type: 'danger'});
```
Comment (danvb112) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectValidation/EditObjectValidation.tsx:107]: Hey @pedro-oliveira446, could you please add a blank line here?

**Review by danvb112 — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectValidation/EditObjectValidation.tsx (hunk 3 above)
Comment (gabrielwas) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectValidation/EditObjectValidation.tsx:107]: Thx Dan

**Review by danvb112 — APPROVED** [review]
LGTM! :tada:

**Comment by danvb112** [comment]
ci:test:sf

**Comment by gabrielwas** [comment]
@pedro-oliveira446 if you try to forward this to Brian, probably we'll have conflicts. Rebase it with Brian's repo.

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Comment by davisaints** [comment]
ci:stable

**Comment by davisaints** [comment]
ci:test:stable

**Comment by pedro-oliveira446** [comment]
Resent [here](https://github.com/brianchandotcom/liferay-portal/pull/141091)

**Pull request closed without merging**